File: L203_Remove_Linked_List_Elements/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head==null) return null;
        
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode pre = dummy;
        while(head !=null){
            if(head.val == val){
                pre.next = head.next;
                head = head.next;

            }else{
                pre =head;
                head = head.next;
                
            }
        }
        return dummy.next;
    }
}

File: L203_Remove_Linked_List_Elements/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        dummy =  ListNode(0)
        dummy.next = head
        pre = dummy
        while head is not None:
            if head.val==val:
                pre.next = head.next
            else:
                pre = head
            
            head = head.next
            
        return dummy.next